src/racer/cargo.rs

```diff
@@ -673,9 +697,18 @@ pub fn get_crate_file(kratename: &str, from_path: &Path) -> Option<PathBuf> {
 
     if let Some(tomlfile) = find_cargo_tomlfile(from_path.to_path_buf()) {
         // look in the lockfile first, if there is one
+        // also search workspaces for a lockfile
         trace!("get_crate_file tomlfile is {:?}", tomlfile);
+
+        let absolute_path = env::current_dir().unwrap().join(from_path);
+        let workspace = find_workspace_root(absolute_path);
+
         let mut lockfile = tomlfile.clone();
         lockfile.pop();
+        if workspace.is_some() {
+            lockfile.push(workspace.unwrap());
```

Comment:
I'm confused what's happening here. It appears lockfile should have a value like `/home/foo/bar/baz/Cargo.lock`. Then, we push the absolute path to a workspace root on the end of it. Assuming `bar` is a workspace root, this would become something like `/home/foo/bar/baz/Cargo.clok/home/foo/bar`. Am I missing something?

Reply:
Unless I'm misunderstanding the docs (https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.push), since `workspace` is an absolute path, it should completely overwrite `lockfile`.

Reply:
Oh wow, that's kind of unexpected! TIL.

I wonder if we could use an assignment instead to make the intent more clear? I worry about others having the same confusion I did when reading the code.

Reply:
Agreed, that would be much better for readability. I thought it was a strange case for `push` as well.